### ISSUE TITLE
loss increases as training progresses

### ISSUE THREAD

**Issue by EpicWayne** [fchollet/keras#1237]
Hi, 
I'm trying to build a bi-directional GRU
 to do gender recognition on text. My model is as follows 

model = Graph()
    model.add_input(name = 'input', input_shape = (None,max_features), dtype = 'float')
    model.add_node(GRU(100, activation='sigmoid', inner_activation='hard_sigmoid'), name='forward', input='input')
    #backwards in time direction
    model.add_node(GRU(100, activation='sigmoid', inner_activation='hard_sigmoid', go_backwards = True), name='backward', input='input')
    model.add_node(Dropout(0.5), name = 'first_dropout', inputs = ['forward','backward'])
    model.add_node(Dense(1, activation='sigmoid'), name='sigmoid', input='first_dropout')
    model.add_output(name='output', input='sigmoid')
    model.compile('SGD',  {'output': 'binary_crossentropy'})

My data is a 3D numpy array (x,y,z) where x is the number of samples, y is the time dimension (Words in the document example) and z is the word2vec encoding of this word (input dimension). The x and y are ints and the z are floats. The labels are binary.

Unfortunately, when I train, the cross entropy loss seems to jump all over the place, and I end up with a test accuracy of 0. Do you have any idea why this is? 
<img width="458" alt="screen shot 2015-12-10 at 8 48 14 am" src="https://cloud.githubusercontent.com/assets/3630063/11716984/67eca744-9f1b-11e5-941c-3ca033196ba4.png">

Thank you very much!

**Comment by rpinsler** [comment]
Did you try to reduce the (default) learning rate of SGD?

**Comment by EpicWayne** [comment]
I have not, I will try that right away. It seems strange though that the model would give me a 0 test accuracy, should a model get approx 0.5 just by random guessing?

**Comment by ylqfp** [comment]
I've also encountered with this problem.
Hoping to get the resolving method.
